src/entrypoints/dead-clicks-autocapture.ts

```diff
@@ -0,0 +1,269 @@
+import { assignableWindow, LazyLoadedDeadClicksAutocapture } from '../utils/globals'
+import { PostHog } from '../posthog-core'
+import { isNull, isNumber, isUndefined } from '../utils/type-utils'
+import { autocaptureCompatibleElements, getEventTarget, isElementNode, isTag } from '../autocapture-utils'
+import { DeadClicksAutoCaptureConfig, Properties } from '../types'
+import { autocapturePropertiesForElement } from '../autocapture'
+import { isElementInToolbar } from '../utils/element-utils'
+
+const DEFAULT_CONFIG: Required<DeadClicksAutoCaptureConfig> = {
+    element_attribute_ignorelist: [],
+    scroll_threshold_ms: 100,
+    selection_change_threshold_ms: 100,
+    mutation_threshold_ms: 2500,
+}
+
+interface Click {
+    node: Element
+    originalEvent: Event
+    timestamp: number
+    // time between click and the most recent scroll
+    scrollDelayMs?: number
+    // time between click and the most recent mutation
+    mutationDelayMs?: number
+    // time between click and the most recent selection changed event
+    selectionChangedDelayMs?: number
+    // if neither scroll nor mutation seen before threshold passed
+    absoluteDelayMs?: number
+}
+
+function asClick(event: Event): Click | null {
+    const eventTarget = getEventTarget(event)
+    if (eventTarget) {
+        return {
+            node: eventTarget,
+            originalEvent: event,
+            timestamp: Date.now(),
+        }
+    }
+    return null
+}
+
+function checkTimeout(value: number | undefined, thresholdMs: number) {
+    return isNumber(value) && value >= thresholdMs
+}
+
+class _LazyLoadedDeadClicksAutocapture implements LazyLoadedDeadClicksAutocapture {
```

Comment:
Nit - I prefer the C# style `ILazy...` to indicate an interface. For our code style maybe 
```suggestion
class LazyLoadedDeadClicksAutocapture implements LazyLoadedDeadClicksAutocaptureInterface {
```

Reply:
I used to make C# interfaces into a sentence `IAmABankAccount` 
So.... careful what you wish for